provider/azure/environ.go

```diff
@@ -1696,13 +1697,13 @@ func (env *azureEnviron) DestroyController(ctx context.ProviderCallContext, cont
 	return nil
 }
 
-func (env *azureEnviron) deleteControllerManagedResourceGroups(ctx context.ProviderCallContext, controllerUUID string) error {
+func (env *azureEnviron) deleteControllerManagedResourceGroups(ctx envcontext.ProviderCallContext, controllerUUID string) error {
 	filter := fmt.Sprintf(
 		"tagName eq '%s' and tagValue eq '%s'",
 		tags.JujuController, controllerUUID,
 	)
 	client := resources.GroupsClient{env.resources}
-	sdkCtx := stdcontext.Background()
+	sdkCtx := context.Background()
```

Comment:
Ultimately we also would want to add Context() to ProviderCallContext just as for BootstrapContext